.github/actions/build_deploy_api/action.yml

```diff
@@ -1,26 +1,28 @@
 name: build-api
-description: Builds the Backend API and Its Docker Image
-inputs:
-  sha_short:
-    description: 'sha_short'
-    required: true
-  new_tag:
-    description: 'new_tag'
-    required: true
-  new_tag_short:
-    description: 'new_tag_short'
-    required: true
-  name:
-    description: 'name'
-    required: true
-  sha:
-    description: "sha"
-    required: true
+on:
+  workflow_call:
+    inputs:
+      sha_short:
+        required: true
+        type: string
+      new_tag:
+        required: true
+        type: string
+      new_tag_short:
+        required: true
+        type: string
+      name:
+        required: true
+        type: string
+      sha:
+        required: true
+        type: string
+
 jobs:
   build-api:
     name: Build and release api
     runs-on: ubuntu-latest
-    if: github.event.inputs.name || github.event.inputs.new_tag
```

Comment:
I think you can still make the `new_tag` optional and refer to it so that we can replace the `github.event`.